lib/ExperimentProvider.js

```diff
@@ -0,0 +1,59 @@
+const seedrandom = require("lib/vendor.bundle").SeedRandom;
+
+exports.ExperimentProvider = class ExperimentProvider {
+  constructor(clientID, experiments) {
+    experiments = experiments || require("experiments.json");
+    this._clientID = clientID;
+    this._rng = seedrandom(clientID);
+    this._data = {};
+    Object.keys(experiments).forEach(key => {
+      const experiment = experiments[key];
+      this.validateExperiment(experiment, key);
+      if (experiment.active === false) {
+        return;
+      }
+      const {id, variant, control} = experiment;
+      const inExperiment = this._rng() <= variant.threshold;
```

Comment:
A single threshold is fine for the single experiment case, but we need a better experiment user selection mechanism to support those multi-experiment scenarios.

For instance, say we have three independent experiments - e1, e2, e3 running at the same time, each of them is targeting 10% of the addon users, respectively. Ideally, the whole user base should be divided into four groups as follows,
* control group, if 0 <= `rng` < 0.7
* e1 group, if 0.7 <= `rng` < 0.8
* e2 group, if 0.8 <= `rng` < 0.9
* e3 group otherwise